### ISSUE TITLE
Should 'pathEq/propEq' and 'pathSatisfies/propSatisfies' parameters' order be consistent

### ISSUE THREAD

**Issue by adispring** [ramda/ramda#2937]
`pathEq`'s path and `propEq`'s prop is the first parameter, but `pathSatisfies` and `propSatisfies` is the second parameter, should their parameters' order be the same order?

**Comment by CrossEye** [comment]
Probably.  I often use `propEq` as my demonstration that although it's usually clear enough what order to put parameters in, once in a while, it simply seems to be an arbitrary decision.  Here these related functions (including `propIs`) seem pretty clear: the predicate/type/value is the thing least likely to change and so should go first.  We should fix `propEq` to match.

Are you interested in writing up a PR for this?


This also shows that we seem to be missing one entry:

|               | Single        |  Multiple     |
|---------------|---------------|---------------|
| **equals**    | propEq        | pathEq        |
| **predicate** | propSatisfies | pathSatisfies |
| **type**      | propIs        |      ??       | 

(Or perhaps that we have one too many.  I don't know if I've *ever* used `propIs` myself.)

**Comment by JamesRamm** [comment]
Just came across this change in the v0.29.0 release. IMO, the predicate/type/value is the *most* likely thing to change and follows the 'data last' philosophy of ramda. 

Here is an example usage of `propEq` for us:
```js
const users = [{ name: 'John',  team: '' }, { name: 'Pete', }, { name: 'Olivia'}, {name: 'Violette'}]
const someRecord = { author: 'Olivia', someProp: 'blah' ;
const findByName = compose(find, propEq('name'));

findByName(someRecord.author)(users);
```

Our usage of `propEq` and `pathEq` is 90% of time geared towards this kind of stuff - filtering lists and such. It is *always* the value that changes, not the property. 

Do you have any other examples or justification of putting the value first?

**Comment by semmel** [comment]
> Just came across this change in the v0.29.0 release. IMO, the predicate/type/value is the most likely thing to change and follows the 'data last' philosophy of ramda.

Yes @JamesRamm, post v0.29 there is some resentment in the air, about this change of parameter order. See also the [recent comments](https://github.com/ramda/ramda/pull/2938#issuecomment-1521946369) in the PR thread.

**Comment by Gastonite** [comment]
It was not a good idea IMO =(

because :
1. It's way more easier to read `propEq('hair', 'brown')` than `propEq('brown', 'hair')`
2. a function `propEq('hair')` is more useful than `propEq('brown')`
3. It creates hard to find bugs
4. It's **just** a cosmetic change that has no other real motivations
5. It introduces a big breaking change in a minor release and does not comply with [semver](https://semver.org/): "MAJOR version when you make incompatible API changes"

Sometime it's better to simply rename a function instead of changing its signature

(I spent 2h to find the origin of bugs in my app)

**Comment by vorg** [comment]
> It was not a good idea to reverse parameters of that function IMO =(
OMG. all my code broke because of this. Why.... :(
